go/fn/object.go

```diff
@@ -759,6 +784,107 @@ func (o KubeObjects) Less(i, j int) bool {
 	return idStrI < idStrJ
 }
 
+func (o KubeObjects) String() string {
+	var elems []string
+	for _, obj := range o {
+		elems = append(elems, strings.TrimSpace(obj.String()))
+	}
+	return strings.Join(elems, "\n---\n")
+}
+
+// Select will return the subset of objects in KubeObjects such that f(object) returns 'true'.
```

Comment:
Naming nit ... Select evokes the SQL "select" statement and lands us in the whole relational-algebra selection vs SQL "where" discussion.

Any alternatives that might avoid this?  `Where` is reasonable, but I'm not sure what the "not" form is - "WhereNot"?  With and Without seem fine.  Exclude is good, but the opposite of Exclude is Include and that doesn't imply removal...

Personally I think `Where` and `WhereNot` might work well, because I'd like us to also be able to do `Where(SomePredicate)` and `Where(Not(SomePredicate))` and then we can say that `WhereNot(SomePredicate)` is a shortcut for `Where(Not(SomePredicate))`

Reply:
I like this new syntax! Thanks a lot for the suggestions @justinsb  
@natasha41575 If we only keep one from the naming candidates `WhereNot` and `Not`, I'd vote for `Not`.  it seems to work better with `Where(Not)` and `Where().Not`.

Reply:
>  If we only keep one from the naming candidates WhereNot and Not, I'd vote for Not. it seems to work better with Where(Not) and Where().Not.

`Not` makes sense. I'm thinking about keeping both; `WhereNot` as a shortcut for `Where(Not)` seems like it can be convenient. My most recent push does this.